### PR TITLE
API / CORS: Use coalesce operator because of falsy values

### PR DESCRIPTION
**Pull request by joselcvarela**
## Scope
While trying to setup `CORS_MAX_AGE: 0` (don't cache) we figured out that this configuration was not being properly applied and the value `undefined` were set instead.
This seems to happen because `0` is a falsy value and the OR `||` operator is being used here:
https://github.com/directus/directus/blob/main/api/src/middleware/cors.ts#L16 

In order to fix this we should use the coalesce operator instead so falsy values can be accepted.

What's changed:

- Accept falsy values for CORS configuration

## Potential Risks / Drawbacks

- CORS is disabled by default (`CORS_ENABLED: false`) but if `CORS_ENABLED: true`, then when `CORS_ORIGIN: false` it will not work as expected, as `false` was not being set and `true` was being used instead: 
 https://github.com/directus/directus/blob/main/api/src/middleware/cors.ts#L11

## Review Notes / Questions

- N/A

### CODE REVIEW

File: api/src/middleware/cors.ts [hunk 1]
```diff
@@ -8,12 +8,12 @@ const env = useEnv();
 
 if (env['CORS_ENABLED'] === true) {
 	corsMiddleware = cors({
-		origin: (env['CORS_ORIGIN'] as string) || true,
-		methods: (env['CORS_METHODS'] as string) || 'GET,POST,PATCH,DELETE',
+		origin: (env['CORS_ORIGIN'] as string) ?? true,
+		methods: (env['CORS_METHODS'] as string) ?? 'GET,POST,PATCH,DELETE',
```
Comment (paescuj) [inline api/src/middleware/cors.ts:12]:
```suggestion
		origin: env['CORS_ORIGIN'] as boolean | string,
		methods: env['CORS_METHODS'] as string,
```

File: api/src/middleware/cors.ts [hunk 2]
```diff
@@ -8,12 +8,12 @@ const env = useEnv();
...
 		allowedHeaders: env['CORS_ALLOWED_HEADERS'] as string,
 		exposedHeaders: env['CORS_EXPOSED_HEADERS'] as string,
-		credentials: (env['CORS_CREDENTIALS'] as boolean) || undefined,
-		maxAge: (env['CORS_MAX_AGE'] as number) || undefined,
+		credentials: (env['CORS_CREDENTIALS'] as boolean) ?? undefined,
+		maxAge: (env['CORS_MAX_AGE'] as number) ?? undefined,
```
Comment (paescuj) [inline api/src/middleware/cors.ts:16]:
```suggestion
		credentials: env['CORS_CREDENTIALS'] as boolean,
		maxAge: env['CORS_MAX_AGE'] as number,
```

**Review by paescuj — COMMENTED** [review]
IMO, the fallback values aren't really necessary at all, since each of these configs have a default value.
@joselcvarela Can you take a look again at my suggestions?

**Review by paescuj — APPROVED** [review]

**Comment by loteoo** [comment]
Bit late to this but I noticed something that might be a bug or an unexpected breaking change. The `CORS_ORIGIN` value used to default to `true`, but since 10.13.0 it now defaults to false. My app only had `CORS_ENABLED=true`, but did not specify `CORS_ORIGIN`. When I upgraded I had CORS errors. So I'm just flagging this in case this was accidental.

**Pull request merged**